### PR TITLE
Add script to generate icons

### PR DESCRIPTION
**Pull request by rehkmansa**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,19 +52,21 @@
...
+    "svg-to-jsx": "^1.0.4",
+    "ts-node": "^10.9.2",
+    "tsx": "^4.19.2",
     "typescript": "^5.6.3",
     "typescript-eslint": "^8.12.2",
     "vitest": "^2.1.4"
-  },
-  "dependencies": {
-    "react": "^18.3.1",
-    "react-dom": "^18.3.1"
```
Comment (ant0n7) [inline package.json]: why did you delete this?

File: scripts/create-icon.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  (props, ref) => {
+    return (
+      ${updatedJsxString}
+    );
+  }
+);
+
+${ComponentName}.displayName = "${ComponentName}";
+
+export default ${ComponentName};`;
```
Comment (ant0n7) [inline scripts/create-icon.ts]: always use named exports

File: scripts/create-icon.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const ComponentName = upperCamelCase(name);
+
+  const jsxString = await svgToJsx(node as string);
+
+  const updatedJsxString = jsxString.replace(
+    /<svg([^>]*)>/,
+    `<svg$1 {...props} ref={ref}>`,
+  );
+
+  const element = `import React, { forwardRef, type SVGAttributes } from "react";
```
Comment (ant0n7) [inline scripts/create-icon.ts]: don't need to import React

File: scripts/create-icon.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  return args
+    .join(" ")
+    .replace(/[^a-zA-Z0-9]+/g, " ")
+    .split(" ")
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+    .join("");
+};
+
+const createElement = async (name: string, node: unknown) => {
+  const ComponentName = upperCamelCase(name);
```
Comment (ant0n7) [inline scripts/create-icon.ts]:
```suggestion
  const componentName = upperCamelCase(name);
```

File: @types/svg-to-jsx.d.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+declare module "svg-to-jsx" {
```
Comment (ant0n7) [inline @types/svg-to-jsx.d.ts]: why is this folder called "@types"? why is it in root?

Comment (ant0n7) [reply @types/svg-to-jsx.d.ts]: can you add a comment about what this does and why this is here. i know it but it should be clearer

File: src/components/icon/icon.test.tsx [hunk 6]
```diff
@@ -1,13 +1 @@
...
-import { Icon } from "./icon";
-import { expect, test } from "vitest";
-
-test("renders icon with default props", () => {
-  render(<Icon />);
-  const iconElement = screen.getByRole("img", { hidden: true });
-  expect(iconElement).toBeInTheDocument();
-  expect(iconElement).toHaveTextContent("icon");
-});
+// TODO: Write tests
```
Comment (ant0n7) [inline src/components/icon/icon.test.tsx]: can you add one so the CI action doesn't fail?

File: src/components/icon/icon.tsx [hunk 7]
```diff
@@ -1,5 +1,21 @@
...
-export const Icon: FC = () => {
-  return <span role="img">icon</span>;
-};
+export interface IconProps extends Omit<SVGAttributes<SVGElement>, "name"> {
+  name: IconKeys;
+  svgName?: string;
+}
+
+export const Icon = forwardRef<SVGSVGElement, IconProps>((props, ref) => {
+  const { name: iconName, svgName: name, ...rest } = props;
```
Comment (ant0n7) [inline src/components/icon/icon.tsx]: why these prop name reassignments? what does svgName do?

File: src/components/icon/icon.tsx [hunk 8]
```diff
@@ -1,5 +1,21 @@
...
+  const Component = Icons[iconName];
+
+  if (!Component) {
+    // TODO: Provide better error message
```
Comment (ant0n7) [inline src/components/icon/icon.tsx]:
```suggestion
    // TODO: Provide a better error message.
```

**Review by ant0n7 — APPROVED** [review]

File: src/components/icon/icon.tsx (hunk 7 above)
Comment (rehkmansa) [reply src/components/icon/icon.tsx]: svg has a prop name by default

File: @types/svg-to-jsx.d.ts (hunk 5 above)
Comment (rehkmansa) [reply @types/svg-to-jsx.d.ts]: I can rename to types

Comment (ant0n7) [reply @types/svg-to-jsx.d.ts]: would be better.
why is it not in source?

File: src/components/icon/icon.tsx (hunk 7 above)
Comment (ant0n7) [reply src/components/icon/icon.tsx]: but why do you rename these props internally? just use them as they are:

```tsx
const Component = Icons[name];
...
<Component ref={ref} name={svgName} {...props} />;
```

consider destructuring within the function args directly instead of a separate line. and ...props instead of ...rest:

```tsx
export const Icon = forwardRef<SVGSVGElement, IconProps>(({name, svgName, ...props}, ref) => {
```

**Review by ant0n7 — APPROVED** [review]
